### PR TITLE
vod doc rough draft init

### PR DESCRIPTION
**Pull request by codentell**
**What does this pull request do? Explain your changes. (required)**

**Specific updates (required)**

## -

- **How did you test each of these updates (required)**

**Does this pull request close any open issues?**

**Screenshots (optional):**

**Checklist:**

- [ ] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by Shih-Yu — COMMENTED** [review]
Looks good for me, waiting to see code review results

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 1]
```diff
@@ -0,0 +1,687 @@
+---
+title: Video On Demand
+description: Livepeer's Video on Demand Guide
+metaTitle: Video on Demand Guide - Livepeer.com
+metaDescription: Livepeer's Video on Demand Guide
+---
+
+# Video on Demand
+
+This is a good tutorial for someone who want to upload video to livepeer using
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: I'd rephrase: This tutorial covers how to upload the video to Livepeer to allow viewers to playback stored video assets in your application.

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 2]
```diff
@@ -0,0 +1,687 @@
+---
+title: Video On Demand
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: Rather than call this Video On Demand, let's call it "Upload a Video" which is clearer to readers -- can you chance the associated URLs, menus, and descriptions to fall along these lines?

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 3]
```diff
@@ -0,0 +1,687 @@
...
+  [Livepeer Video Service API key](https://livepeer.com/docs/guides/start-live-streaming/api-key)
+
+## Assets
+
+When providing an asset for VOD, there are two options, either
+
+1. Import the asset - **only MP4 file format**
+2. Upload the asset - gets transcoded to MP4(H.264 + AAC)
+
+### **Importing assets**
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: We've removed the distinction between "Importing" and "Uploading" assets in the API, changes will be merged 5/31/22. They can be renamed "Upload from URL" and "Upload Locally".

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 4]
```diff
@@ -0,0 +1,687 @@
...
+  - Using yarn
+  ```bash
+  yarn add axios
+  ```
+- Get a
+  [Livepeer Video Service API key](https://livepeer.com/docs/guides/start-live-streaming/api-key)
+
+## Assets
+
+When providing an asset for VOD, there are two options, either
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]:
```
"There are two methods to upload a video to Livepeer:

- Upload via URL, with an .mp4 file format
- Upload locally" 
```

Comment (pglowacky) [reply packages/www/docs/guides/video-on-demand/index.mdx]: To clarify, both uploading via URL and uploading locally will be transcoded. The way it reads is that only the local upload will be transcoded.

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 5]
```diff
@@ -0,0 +1,687 @@
...
+      "updatedAt": 1652894665451
+    }
+  }
+}
+```
+
+> _Note that the status is pending, it might take a few minutes for it to be
+> imported, to verify use
+> [`https://livepeer.com/api/asset/$ASSET_ID/`](https://livepeer.com/api/asset/$ASSET_ID/)
+> with provided [asset.id](http://asset.id) in the response from above_
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: Is this clear enough? Should we add code that developers can copy and paste more easily?

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 6]
```diff
@@ -0,0 +1,687 @@
...
+  "downloadUrl": "https://livepeercdn.com/asset/8728n2iaziowj67d/video"
+}
+```
+
+### **Uploading assets**
+
+When using the upload asset method it will transcode the video, it is a two-step
+process:
+
+> _Only MP4 files encoded with H.264 and AAC are supported at the moment_
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: remove `at the moment`

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 7]
```diff
@@ -0,0 +1,687 @@
...
+
+> At the moment o*nly videos uploaded in landscape mode will return a*
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: Clean up formatting -- there's an asterisk in `only`, remove `at the moment`

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 8]
```diff
@@ -0,0 +1,687 @@
...
+> `playbackUrl`
+
+> _File size limit under 1Gb is fine, anything greater will time out with
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: `File size is limited to 1GB. Any files greater than 1GB will error with `NETWORK CHANGED` `

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 9]
```diff
@@ -0,0 +1,687 @@
...
+> NETWORK CHANGED error in API_
+
+**Step 1**
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: use H2 header for better formatting

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 10]
```diff
@@ -0,0 +1,687 @@
...
+
+Generate the URL to upload assets
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: Should be H3 header or bolded

File: packages/www/docs/guides/video-on-demand/index.mdx (hunk 9 above)
Comment (pglowacky) [reply packages/www/docs/guides/video-on-demand/index.mdx]: May want to clean up headers throughout document to improve readability

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 11]
```diff
@@ -0,0 +1,687 @@
...
+    }
+  }
+}
+```
+
+**Step 2**
+
+Upload the asset to the generated URL
+
+- Get the `“url”: “https://origin.livepeer.com/api/assets/…“` from the response
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: clean up formatting

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 12]
```diff
@@ -0,0 +1,687 @@
...
+  .then((res) => {
+    console.log(JSON.stringify(res.data));
+  })
+  .catch((error) => console.log(error));
+```
+
+Response
+
+- When the video is uploaded, it should return a 200 response
+- Use the `*https://livepeer.com/api/asset/$ASSET_ID/` with the provided
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: Clean up formatting

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 13]
```diff
@@ -0,0 +1,687 @@
...
+  },
+  "playbackId": "4c21k7z2tjcdg8wt",
+  "playbackUrl": "https://livepeercdn.com/recordings/54aeff83-404c-490a-9afa-99296a932401/index.m3u8",
+  "downloadUrl": "https://livepeercdn.com/asset/4c21k7z2tjcdg8wt/video"
+}
+```
+
+### **List Assets**
+
+When listing assets, look for the **status** in the response body, it should
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: Wondering if we should call out that we're working on webhooks for this or suggest polling. Seems odd to suggest to the reader to simply wait or manually try again.

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 14]
```diff
@@ -0,0 +1,687 @@
...
+  },
+  "userId": "eada599f-3d58-499b-ba24-c7f3faf988de",
+  "createdAt": 1652903903963,
+  "inputAssetId": "87289307-b60f-432b-984a-8ede174fa152"
+}
+```
+
+### **List Tasks**
+
+When listing tasks, it will provide information on imported or exported
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: What's "it" in this context? The API?

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 15]
```diff
@@ -0,0 +1,687 @@
...
+information on the asset
+
+Use the list tasks method to get CID and metadata -
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: remove the em dash

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 16]
```diff
@@ -0,0 +1,687 @@
...
+
+To Get the CID and metadata URL for the asset that was just exported,  you can
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: This paragraph should be rewritten for clarity. Feel free to reach out to me for suggestions.

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 17]
```diff
@@ -0,0 +1,687 @@
...
+  },
+  "userId": "eada599f-3d58-499b-ba24-c7f3faf988de",
+  "createdAt": 1652894665451,
+  "outputAssetId": "87289307-b60f-432b-984a-8ede174fa152"
+}
+```
+
+Caveats
+
+- At the moment, cannot edit or delete assets
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: This isn't true and needs to be added. @codentell can you reach out to @victorges to get these end points documented?

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 18]
```diff
@@ -0,0 +1,687 @@
...
+    "phase": "completed",
+    "updatedAt": 1652894675639
+  },
+  "userId": "eada599f-3d58-499b-ba24-c7f3faf988de",
+  "createdAt": 1652894665451,
+  "outputAssetId": "87289307-b60f-432b-984a-8ede174fa152"
+}
+```
+
+Caveats
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: Perhaps this section goes at the top of the article? These are good things to know before beginning to work with our VOD API

File: packages/www/docs/guides/video-on-demand/index.mdx [hunk 19]
```diff
@@ -0,0 +1,687 @@
...
+
+- At the moment, cannot edit or delete assets
+- Have to manually check the status of the file using `List Task` method to see
+  if it has finished uploading
+- Only videos uploaded in landscape mode return an HSL playback URL at the
+  moment
+- No limit to the size of assets uploaded but will time out with anything
+  greater than 1Gb
+- Only assets that use the Upload method will be transcoded
```
Comment (pglowacky) [inline packages/www/docs/guides/video-on-demand/index.mdx]: @victorges is this true? I don't think it is?

**Review by pglowacky — COMMENTED** [review]
High level feedback:

- Overall, it covers the right topics. 
- Needs some formatting work to make it more readable, including proper use of headers. 
- There are some pretty blatant copywriting errors that should've gotten caught in a read-through. At least one section needs to be rewritten for clarity. 
- I think we should add that we're working on webhooks (or include instructions for polling)  to understand the status of the tasks more manageable. It feels a little strange to suggest that users keep manually trying. 
- Both methods of uploading are transcoded now

File: packages/www/docs/guides/video-on-demand/index.mdx (hunk 13 above)
Comment (Shih-Yu) [reply packages/www/docs/guides/video-on-demand/index.mdx]: At the time, this was the suggestion, but notifying a webhook is being worked on would help the developer

File: packages/www/docs/guides/video-on-demand/index.mdx (hunk 1 above)
Comment (codentell) [reply packages/www/docs/guides/video-on-demand/index.mdx]: Shih-yu resolved changes

File: packages/www/docs/guides/video-on-demand/index.mdx (hunk 2 above)
Comment (codentell) [reply packages/www/docs/guides/video-on-demand/index.mdx]: Shih-yu resolved changes

File: packages/www/docs/guides/video-on-demand/index.mdx (hunk 3 above)
Comment (codentell) [reply packages/www/docs/guides/video-on-demand/index.mdx]: Shih-yu resolved changes

File: packages/www/docs/guides/video-on-demand/index.mdx (hunk 4 above)
Comment (codentell) [reply packages/www/docs/guides/video-on-demand/index.mdx]: Shih-yu resolved changes

File: packages/www/docs/guides/video-on-demand/index.mdx (hunk 7 above)
Comment (codentell) [reply packages/www/docs/guides/video-on-demand/index.mdx]: Shih-yu resolved changes

File: packages/www/docs/guides/video-on-demand/index.mdx (hunk 6 above)
Comment (codentell) [reply packages/www/docs/guides/video-on-demand/index.mdx]: Shih-yu resolved changes

File: packages/www/docs/guides/video-on-demand/index.mdx (hunk 8 above)
Comment (codentell) [reply packages/www/docs/guides/video-on-demand/index.mdx]: Shih-yu resolved changes

File: packages/www/docs/guides/video-on-demand/index.mdx (hunk 9 above)
Comment (codentell) [reply packages/www/docs/guides/video-on-demand/index.mdx]: Shih-yu resolved changes

File: packages/www/docs/guides/video-on-demand/index.mdx (hunk 10 above)
Comment (codentell) [reply packages/www/docs/guides/video-on-demand/index.mdx]: Shih-yu resolved changes

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 20]
```diff
@@ -0,0 +1,682 @@
...
+---
+
+# Upload a Video
+
+This tutorial covers how to upload the video to Livepeer to allow viewers to
+playback stored video assets in your application.
+
+## Caveats
+
+- Have to manually check the status of the file using `List Task` method to see
```
Comment (pglowacky) [inline packages/www/docs/guides/upload-a-video/index.mdx]: I still think this is a funny thing to ask a developer to do -- we should either tell readers that we're currently working on webhooks (which feels like the obvious thing that users are going to want) or give them instructions to do polling. @victorges thoughts on this?

If we decide that this okay, it needs to be rephrased: `Currently, users need to manually check the status of the file using List Task method (described below) to see if uploads have been completed. We're currently working on webhooks to make this easier for users.`

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 21]
```diff
@@ -0,0 +1,682 @@
...
+```
+
+### **List Tasks**
+
+When listing tasks, the response will provide information on imported or
+exported information on the asset
+
+Use the list tasks method to get CID and metadata
+
+To Get the CID and metadata URL for the asset that was just exported,  you can
```
Comment (pglowacky) [inline packages/www/docs/guides/upload-a-video/index.mdx]: Change to 

To get the CID and metadata URL for the new, just created asset, retrieve the task until it is completed. To do this, check the **status** field in the response. This finished task will have an output with all the information about that asst, and the CID information is located under the `output.export.ipfs` path.

**Review by pglowacky — COMMENTED** [review]
@Shih-Yu @codentell Please re-read through and make punctuation consistent. There are lots of unpunctuated sentences. Feel free to ask @hnjolles1 if you're unsure about any copywriting details.

**Comment by pglowacky** [comment]
How difficult is it to have the side bar menu reflect the major sections in this doc? Could make it easier for developers to find what they're looking for or to figure out if we have the right functionality.

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 22]
```diff
@@ -0,0 +1,775 @@
...
+title: Upload a Video
+description: Livepeer's Upload a Video Guide
+metaTitle: Upload a Video Guide - Livepeer.com
+metaDescription: Livepeer's Upload a Video Guide
+---
+
+# Upload a Video
+
+This tutorial covers how to upload the video to Livepeer to allow viewers to
+playback stored video assets in your application.
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]: I think the writing could be greatly improved here. The [Mux API documentation](https://docs.mux.com/guides/video/stream-video-files) is often a great resource on how we could present some of these concepts, and is in fact the inspiration for many of our doc content here.

I believe you should also think of this as an introduction to the VOD API in general, since this will be the only non-API reference piece in our docs for now.

As an example, I think this will make this introduction cleaner and clearer:

```suggestion
The Video-on-Demand (VOD) API allows you to send video files to Livepeer and get them ready for optimized playback. This tutorial covers how to use it to upload videos and play them back in your application. 

Those videos can be provided either by you (static content) or your users, given your application offers an interface for them to do so.
```

File: packages/www/docs/guides/upload-a-video/index.mdx (hunk 20 above)
Comment (victorges) [reply packages/www/docs/guides/upload-a-video/index.mdx]: Agreed! I don't think this is a caveat worth noting here as a big issue in the flow. It's just how things work for now while we don't have a more sophisticated interface. Webhooks will be nicer for communicating the state to a backend, but it won't make much of a difference for users building only a frontend (or simpler backends without custom state) anyway.

Also agree that we **must** provide instructions for polling along this guide. Can also link to the `video-nft` SDK (that needs rebranding) for an example on how to wait for a task to finish: https://github.com/livepeer/video-nft/blob/4e24bd46a884f7a0c23fe23b233cfd0c9a3bafad/src/minter.ts#L480

We can also create a simple `gist` or code snippet for that using axios.

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 23]
```diff
@@ -0,0 +1,775 @@
...
+
+## Caveats
+
+- Have to manually check the status of the file using `List Task` method to see
+  if it has finished uploading.
+- File size is limited to 1GB. Any files greater than 1GB will error with
+  NETWORK CHANGED.
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]: I'm not aware of this error and do not believe that it will always error out with that. Would rephrase with
```suggestion
- Files are currently limited to 1GB in size. Any files greater than that will likely error out during the upload or processing steps.
```

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 24]
```diff
@@ -0,0 +1,775 @@
...
+  - Using npm
+  ```bash
+  npm axios
+  ```
+  - Using yarn
+  ```bash
+  yarn add axios
+  ```
+- Get a
+  [Livepeer Video Service API key](https://livepeer.com/docs/guides/start-live-streaming/api-key)
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]: I think all these steps are unnecessary. This is a technical documentation for developers getting started with the VOD API. If they are using JavaScript, they already have `node` and a package manager and know how to use it. I'd just make a remark here that we will be using Node and `axios` so they can install it on their projects (IMO don't need to teach them how to do it either).

This guide is also inside the "Livepeer Video Services" website, so we should not keep repeating that. It's very verbose/cumbersome to read so it's better to avoid it.

```suggestion
## Dependencies

In this tutorial, we will be using [Node.js](https://nodejs.org/en/) and [Axios](https://axios-http.com/) so make sure to set that up in your project. You will also need to [Get an API key](https://livepeer.com/docs/guides/start-live-streaming/api-key) to call the VOD API.
```

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 25]
```diff
@@ -0,0 +1,775 @@
...
+
+## Assets
+
+There are two methods to upload a video to Livepeer:
+
+- Upload via URL, with an `.mp4` file format
+
+- Upload locally
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]: We only support MP4 everywhere, and the caveat has already been made in the beginning. No need to reemphasize it here and only in one of the methods.

Can also improve the wording:
```suggestion
- Import from a URL

- Directly upload from a local file
```
(we theoretically wanted to use "upload" on both cases but I don't really think it makes sense... The API is still called `/import` anyway so it will be confusing not to use that here. @pglowacky do you really think we need to call it upload here? If so let me know and I can rename the API as well)

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 26]
```diff
@@ -0,0 +1,775 @@
...
+
+## Assets
+
+There are two methods to upload a video to Livepeer:
+
+- `Upload via URL`, with an `.mp4` file format
+
+- `Upload locally`
+
+### Upload from URL
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]: Should be consistent with how we call the steps in the beginning here

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 27]
```diff
@@ -0,0 +1,775 @@
...
+  npm axios
+  ```
+  - Using yarn
+  ```bash
+  yarn add axios
+  ```
+- Get a
+  [Livepeer Video Service API key](https://livepeer.com/docs/guides/start-live-streaming/api-key)
+
+## Assets
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]: I don't think this section name makes sense.

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 28]
```diff
@@ -0,0 +1,775 @@
...
+- `Upload via URL`, with an `.mp4` file format
+
+- `Upload locally`
+
+### Upload from URL
+
+When using the `upload via URL` method:
+
+- Provide the name of the asset
+- Provide the URL link of the asset that is to be downloaded
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]:
```suggestion
- Provide the URL of the asset that should be imported
```

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 29]
```diff
@@ -0,0 +1,775 @@
...
+
+JS + axios
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]: I would just call this `JavaScript`. We already made the disclaim in the beginning that we will be using `axios`.

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 30]
```diff
@@ -0,0 +1,775 @@
...
+
+```jsx
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]: Isn't this just `js`?

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 31]
```diff
@@ -0,0 +1,775 @@
...
+    data: JSON.stringify({
+      "url": "$EXTERNAL_URL",
+      "name": "Example name"
+    })
+  }
+})
+.then(res => {
+  console.log(JSON.stringify(res.data))
+})
+.catch(error => console.log(error))
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]: I think we can provide some cleaner code here using `async/await` constructs. Can make it a function if you're worried users will try to run it on a non-async-await environment. Applies to all other JS snippets.

```suggestion
const axios = require('axios');

async function importAsset(name, url) {
	const res = await axios.post('https://livepeer.com/api/asset/import', {
		headers: {
			Authorization: 'Bearer {api_key}',
			'Content-Type': 'application/json'
		},
		data: JSON.stringify({ url, name })
	});
	const { asset, task } = res.data;
	return { asset, task };
}

importAsset('Example name', '$EXTERNAL_URL').then(console.log, console.error);
```

Also, might omit the `require('axios')` as well if we can assume users will already do it only once in their source files.

Comment (victorges) [reply packages/www/docs/guides/upload-a-video/index.mdx]: Actually, this doesn't even "compile"...
Screen Shot 2022-06-01 at 20 41 35

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 32]
```diff
@@ -0,0 +1,775 @@
...
+        "url": "https://www.pexels.com/video/6394054/download/"
+      }
+    },
+    "status": {
+      "phase": "pending",
+      "updatedAt": 1652894665451
+    }
+  }
+}
+```
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx:132]: These example JSONs are taking most of the space in the guide. I'm not sure if it's worth including them as they don't add a lot of useful information.

These probably belong in the API reference instead. I'd say you can move the examples to the corresponding API reference and link to the reference doc here.

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 33]
```diff
@@ -0,0 +1,775 @@
...
+
+### Upload Locally
+
+When using the `upload locally` method, this is a two-step process:
+
+> _Only MP4 files encoded with H.264 and AAC are supported_
+
+> _File size limited to 1GB. Any files greater than 1GB will error with NETWORK
+> CHANGED_
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]: We already have both of these in the beginning of the doc.

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 34]
```diff
@@ -0,0 +1,775 @@
...
+    ],
+    "duration": 12.031667
+  },
+  "playbackId": "8728n2iaziowj67d",
+  "playbackUrl": "https://livepeercdn.com/recordings/e605988e-e8d3-4e14-b5b0-10b01ff94a8b/index.m3u8",
+  "downloadUrl": "https://livepeercdn.com/asset/8728n2iaziowj67d/video"
+}
+```
+
+### Upload Locally
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx:190]: Should also match the enumeration of the method in the beginning, if you change that.

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 35]
```diff
@@ -0,0 +1,775 @@
...
+      "updatedAt": 1652894665451
+    }
+  }
+}
+```
+
+> _Note that the status is `pending`, it might take a few minutes for it to be
+> imported. To verify that the status is `ready`, use
+> `https://livepeer.com/api/asset/$ASSET_ID/` with the provided `asset.id` in
+> the response above_.
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx:137]: This is a crucial step of the "adding a video file" process and should not be just a note, as Paige suggested.

Ideally we should also provide a snippet on how to wait for a task, which can be copied (and adapted) from the `video-nft` SDK. Bonus if it can also have the `reportProgress` function which can be used to update an UI with the progress etc.

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 36]
```diff
@@ -0,0 +1,775 @@
...
+
+Response
+
+```json
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx:141]: Same, I don't think these are worth it.

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 37]
```diff
@@ -0,0 +1,775 @@
...
+      "Content-Type": "application/json",
+    },
+    data: JSON.stringify({
+      name: "Example name",
+    }),
+  })
+  .then((res) => {
+    console.log(JSON.stringify(res.data));
+  })
+  .catch((error) => console.log(error));
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]: This one compiles 😄 

I think it would be better using async/await as well though.

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 38]
```diff
@@ -0,0 +1,775 @@
...
+    }
+  }
+}
+```
+
+## Step 2
+
+Upload the asset to the generated URL
+
+- Get the `“url”: “https://origin.livepeer.com/api/assets/…“` from the response.
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx:259]: The URL will not always be like this. The user should not have any logic regarding what that URL should look like, they should just use it as a destination for the `PUT` with the file content. As such, I wouldn't include the URL format in this instruction.

```suggestion
- Get the `url` field from the response above.
```

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 39]
```diff
@@ -0,0 +1,775 @@
...
+  ```json
+  {
+    "id": "4c21895e-2488-4c0a-9377-3369ff586ddd",
+    "name": "Waterfall",
+    "status": "waiting",
+    "userId": "eada599f-3d58-499b-ba24-c7f3faf988de",
+    "createdAt": 1652895834393,
+    "playbackId": "4c21k7z2tjcdg8wt"
+  }
+  ```
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx:273]: This is not necessary here.
```suggestion
```

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 40]
```diff
@@ -0,0 +1,775 @@
...
+
+- Upload the video
+
+JS + axios
+
+```jsx
+// 2) Upload the contents
+
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]:
```suggestion
```

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 41]
```diff
@@ -0,0 +1,775 @@
...
+JS + axios
+
+```jsx
+const axios = require("axios");
+
+axios
+  .post("https://livepeer.com/api/asset/request-upload", {
+    headers: {
+      Authorization: "Bearer {api_key}",
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]: Will be better to use actual JavaScript syntax here that users can simply add some env var later or something like that to make it work. e.g.:
```suggestion
      Authorization: `Bearer ${process.env.LP_API_KEY}`,
```

Applies to all other snippets using API keys.

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 42]
```diff
@@ -0,0 +1,775 @@
...
+  .put("${url}", {
+    headers: {
+      "Content-Type": "video/mp4",
+    },
+    data: "@$VIDEO_FILE_PATH",
+  })
+  .then((res) => {
+    console.log(JSON.stringify(res.data));
+  })
+  .catch((error) => console.log(error));
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]: These code snippets will be much more useful if they build on top of each other. So instead of building independent snippets that can only be run like a script that gets all input from templated vars, the first snippet in this process should populate an `url` variable which is then used here.

The video file should also not be sent as a string. If the idea is to read from a file, you should include here the code that opens a file and sends it as the `axios` request payload. This can be as simple as `fs.createReadStream(path)` and sending that as the request `data`.

Also same regarding using async/await for cleanliness.

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 43]
```diff
@@ -0,0 +1,775 @@
...
+```
+
+Response
+
+- When the video is uploaded, the result should return a 200 response
+- Use the `https://livepeer.com/api/asset/$ASSET_ID/` with the provided
+  asset.id\* to check that the video is uploaded, the status should be
+  `"status": "ready"`
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]: Do not need to repeat this on both methods of uploading a file.

IMO should instead: 
 - introduce the "asset" and "task" resource types in the beginning of the documentation (around the current `## Assets` section)
 - Explain that adding a video file is an asynchronous process that requires processing the file. Because of that, we will first explain how to send the data to the API and then how to track the progress of the processing
 - in each of the methods, specify _when_ should the code grab the `asset` and `task` references/IDs (on the response of `/import` and `/request-upload`)
 - After both methods, explain how to monitor the progress of the tasks and then grab the final imported asset.

File: packages/www/docs/guides/upload-a-video/index.mdx [hunk 44]
```diff
@@ -0,0 +1,775 @@
...
+  },
+  "status": {
+    "phase": "completed",
+    "updatedAt": 1652894675639
+  },
+  "userId": "eada599f-3d58-499b-ba24-c7f3faf988de",
+  "createdAt": 1652894665451,
+  "outputAssetId": "87289307-b60f-432b-984a-8ede174fa152"
+}
+```
```
Comment (victorges) [inline packages/www/docs/guides/upload-a-video/index.mdx]: I don't think any of these belong in this guide. We can create proper API references for those (including request/response examples) in the API reference (and maybe link to them here), but they should not be part of a "upload a video" tutorial (which is also the only VOD guide).

I think all this should be replaced with the part of the guide that teaches how to monitor the progress of an upload instead.

**Comment by pglowacky** [comment]
Publishing for now (I keep getting questions about VOD and desperately need to be able to send documentation). 

@Shih-Yu @codentell Have all of Victor's comments been addressed?

**Comment by Shih-Yu** [comment]
> Publishing for now (I keep getting questions about VOD and desperately need to be able to send documentation).
> 
> @Shih-Yu @codentell Have all of Victor's comments been addressed?

@pglowacky Only these two comments still need to be addressed: 
https://github.com/livepeer/livepeer-com/pull/1074#discussion_r887368397
https://github.com/livepeer/livepeer-com/pull/1074#discussion_r887377636
